### PR TITLE
update jwt-go to v3.0.0-4-g01aeca5

### PR DESCRIPTION
**Pull request by colemickens**
[Analytics]()
**update jwt-go to v3.0.0-4-g01aeca5**
new jwt-go rev: 01aeca54ebda6e0fbfafd0a524d234159c05ec20

---

I need this to add Azure cloudprovider since the Azure SDK uses the newer JWT library (and the API changed).

Also relevant for https://github.com/kubernetes/kubernetes/issues/28180.

### CODE REVIEW

**Comment by liggitt** [comment]
Can you split the non godeps changes into their own commit?

**Comment by colemickens** [comment]
Split.

The first commit (`godep: update jwt-go to v3.0.0-4-g01aeca5`) contains the change to `Godeps.json` and the changes to `vendor`.

The second commit (`fix serviceaccount's use of jwt-go`) contains the trivial changes needed due to jwt-go's API changes.

**Comment by liggitt** [comment]
thanks, can you tweak the commit title of the bump commit to `bump(github.com/dgrijalva/jwt-go): 01aeca54ebda6e0fbfafd0a524d234159c05ec20`? makes it easier to find godep bumps in git logs

LGTM otherwise

**Comment by colemickens** [comment]
@liggitt updated, PTAL. Thanks!

**Comment by liggitt** [comment]
LGTM, thanks

**Comment by liggitt** [comment]
@k8s-bot ok to test

**Comment by liggitt** [comment]
ok... the bot has lost its mind

**Comment by liggitt** [comment]
@k8s-bot test this

**Comment by ixdy** [comment]
@k8s-bot test this please, issue #27462

**Comment by liggitt** [comment]
@k8s-bot unit test this please, issue #27462

**Comment by wojtek-t** [comment]
@k8s-bot test this please, issue #27462

File: pkg/serviceaccount/jwt.go [hunk 1]
```diff
@@ -133,6 +135,8 @@ func (j *jwtTokenAuthenticator) AuthenticateToken(token string) (user.Info, bool
 			return key, nil
 		})
 
+		claims, _ := parsedToken.Claims.(jwt.MapClaims)
```
Comment (cjcullen) [inline pkg/serviceaccount/jwt.go]: This needs to come after the error check. Otherwise we panic if we ever have a token that isn't a jwt.

Comment (liggitt) [inline pkg/serviceaccount/jwt.go]: good catch, missed that. @colemickens, can you open a follow up?

Comment (liggitt) [inline pkg/serviceaccount/jwt.go]: actually, I opened https://github.com/kubernetes/kubernetes/pull/28542 with a test for the invalid token case

**Pull request merged**